api/src/main/java/io/druid/data/input/impl/FileIteratingFirehose.java

```diff
@@ -44,13 +47,26 @@ public FileIteratingFirehose(
   {
     this.lineIterators = lineIterators;
     this.parser = parser;
+    final ParseSpec parseSpec = parser.getParseSpec();
+    if (parseSpec instanceof CSVParseSpec) {
```

Comment:
If we decide that Parsers return nulls or ParseResults for signalling "skip me", they should be able to decide the ```skipHeaderRows``` option is effective or not. This means, they should be able to be aware of which types of index tasks are calling them. I think this requires to add an interface to ParseSpec, but ```skipHeaderRows``` is only valid for text formats which lines are terminated by ```\n```, which makes difficult to add an abstract interface for all types of ParseSpecs. So, I think the index task calling Parser should check that the Parser supports ```skipHeaderRows``` by using instanceof, and reset it if the index task is not a non-Hadoop batch index task.